homeassistant/components/sensor/__init__.py

```diff
@@ -476,6 +475,18 @@ def state(self) -> Any:
                     f"Sensor {self.entity_id} provides state value '{value}', "
                     "which is not in the list of options provided"
                 )
+            return value
+
+        # Other device classes should have a valid numeric value
+        if not isinstance(value, (int, float, Decimal)):
+            try:
+                value = float(value)  # type: ignore[arg-type]
```

Comment:
This will result in a slight change of behavior, as "1" will get converted to "1.0"
We could improve with the code below, but then "1.0" will get converted to "1"
```python
                if value.is_integer():
                    value = int(value)
```

Based on how the tests behave on the full suite, I would suggest that we standardize and always remove decimals if they are equal to zero.

Reply:
I have cancelled this change.
For now we only validate that it could be converted into a float.

Maybe we can implement standardization in a follow-up PR.
Also see https://github.com/home-assistant/frontend/pull/14927